### ISSUE TITLE
Make solution compile-able on MacOS

### ISSUE THREAD

**Issue by linkdotnet** [bitfoundation/bitplatform#2871]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

The current solution is in .net6.0, but it cannot be built on MacOS (and presumably Linux).

The reason for this is the package [`BuildWebCompiler2022`](https://www.nuget.org/packages/BuildWebCompiler2022) which is only available in .netcoreapp3.1 and not .net6.0.
Furthermore, the package tries hard to access "cmd.exe", which of course is not available on MacOS, nor on Linux.

### Describe the solution you'd like

The solution would replace `BuildWebCompiler2022` with platform-independent solution.

### Additional context

<img width="1436" alt="image" src="https://user-images.githubusercontent.com/26365461/182546866-abcaf477-84de-4b2e-8178-7a3442f4bae1.png">

**Comment by msynk** [comment]
@linkdotnet thanks for submitting this issue.
it seems the nuget package that we are using for compiling .scss files (BuildWebCompiler2022) is only for Windows. we need to investigate it and find a proper solution.

**Comment by linkdotnet** [comment]
Maybe a few ideas from my side. `AntBlazor` has a similiar way of dealing with things. They internally have their own [`AntDesignTestBase`]( https://github.com/ant-design-blazor/ant-design-blazor/blob/master/tests/AntDesign.TestKit/AntDesignTestBase.cs). For BitPlatform "we" could introduce a similiar approach. For example a dedicated nuget package for bUnit which makes testing easier. I am not aware of any other unit testing framework for Blazor, so that should keep maintenance cost low.

Additionally one could introduce extension methods for clicking a button for example:
```csharp
public class BitButtonTestExtensions
{
    public static void ClickButton(this IRenderFragment<BitButton> button)
    {
        if (button.Href is not null) 
            button.Click("a");
        else
            button.Click("button");
    }
}
```

The user can then simply use the extensions:
```csharp
// By inheriting from BitPlatformTestContext all JS is already either setup or JSRuntime is set to Loose
public class MyNiceTest : BitPlatformTestContext
{
    [Fact]
    public void ShouldDoAwesomeStuff()
    {
        var cut = RenderComponent<MyCustomComponent>();

        // Now we can use the extension method
        cut.FindComponent<BitButton>().ClickButton();

        // Assert
    }
}
```

Paired with a nice documentation that should cover a lot of use cases. Keep in mind that a simple documentation might be good enough for starters and if you or we from bUnit see users often time running into trouble those helpers might be useful.

Let me know your thoughts.

**Comment by ysmoradi** [comment]
Part of the issue has been resolved by #2893

**Comment by linkdotnet** [comment]
Small input from my side as I fiddled around on my MAC:
This seems a potential candidate: https://github.com/koenvzeijl/AspNetCore.SassCompiler

The problem with LibSassBuilder (cc @ysmoradi) is that it does not support arm64 architecture (issue gets tracked [here](https://github.com/johan-v-r/LibSassBuilder/issues/48)). There is a newer version [DartSassBuilder](https://github.com/deanwiseman/DartSassBuilder) but this seems to have a high risk of not being maintained very long.

**Comment by ysmoradi** [comment]
We rely on `BuildWebCompiler2022+` in the following scenarios:
1- Building CSS files from SASS files
2- Minifying JavaScript output of the TypeScript compiler
To make the solution compilable on Mac / Linux, we need replacement(s) for both usages

-----

At the moment, we've removed javascript minification from our project templates (TodoTemplate and AdminPanel), and we've replaced `BuildWebCompiler2022+` with `LibSassBuilder` so we do expect people would be able to build our project templates on all operating systems.

-----

Note that ProjectTemplates and even Bit Blazor UI demo project (Called Playground) have an App project which is a blazor hybrid project and at the moment, you've to unload that and we don't have a short-term plan to support blazor hybrid in IDE/Platform other than VIsual Studio / Windows

-----

P.S.
It seems that LibSassBuilder itself won't work on new mac devices as you mentioned before.
I can create a temporary branch that contains project templates with DartScssBuilder so we can make sure that project templates are good to go

-----

For Bit Blazor UI and its demo project, we need to figure out what is/are replacement(s) for `BuildWebCompiler2022+` that fulfill our requirements (javascript minification and SASS to CSS compilation)

@linkdotnet

**Comment by linkdotnet** [comment]
Thanks for the input @ysmoradi.

Please don't consider this with urgency. Seems like not that many people have that problem and for my part, I do have a spare windows notebook. So that is more of a luxury problem ;)

**Comment by ysmoradi** [comment]
Note that projects that are being created using bit project templates are now compilable on all platforms thanks to `Abstractions Sass Compiler` nuget package
Need more work on Bit Blazor UI & its demo project

**Comment by ysmoradi** [comment]
Closed by #4529
